src/starknetCli.ts

```diff
@@ -200,7 +206,11 @@ export async function runStarknetDeploy(filePath: string, options: IDeployProps)
     const classHash = compileResult.classHash;
     execSync(
       `${warpVenvPrefix} starknet deploy --network ${options.network} ${
-        options.no_wallet ? `--no_wallet --contract ${resultPath} ` : `--class_hash ${classHash}`
+        options.no_wallet
+          ? `--no_wallet --contract ${resultPath} `
+          : options.wallet === undefined
+          ? `--class_hash ${classHash}`
```

Comment:
(Related to previous comment)
And then you could scrap this  extra ternary (and any extra ternary of the future that needs the `wallet` option). For this to be fully functional you will need to verify at the beginning that `options.no_wallet || option.wallet !==undefined` expression is True

Reply:
Also nitpick:  Should we aloud the `options.no_wallet` and the `option.wallet` to be passed at the same time. Perhaps modifying the previous boolean expression to using a xor like `options.no_wallet xor option.wallet !==undefined`